service/service.go

```diff
@@ -322,10 +346,15 @@ func (app *Application) shutdownPipelines() {
 	app.builtReceivers.StopAll()
 
 	app.logger.Info("Stopping processors...")
-	app.builtPipelines.ShutdownProcessors(app.logger)
+	err := app.builtPipelines.ShutdownProcessors(app.logger)
+	if err != nil {
```

Comment:
This one is tricky: we previously ignored the error, which means that we proceeded to `ShutdownAll`.
What would be the correct behavior?

Reply:
Also I've noticed that currently we return error as soon as one processor failed to shutdown:
https://github.com/open-telemetry/opentelemetry-collector/blob/91728bc8ce02f78886a52303de7509a71c1fee29/service/builder/pipelines_builder.go#L65-L69

This seems like a bug to me, we should proceed to shutdown remaining processors instead, and return collected errors at the very end.

Reply:
See #628

Reply:
Fixed.